pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -86,7 +86,10 @@ func (u UsernamePassword) Setup(baseDir string) error {
 	fmt.Fprintf(gitconfig, UserPassGitConfig, gitcredentials.Name())
 	fmt.Fprintf(gitcredentials, "%s", usernamePasswordURL.String())
 
-	return ensureGitConfigIncludes(gitconfig.Name())
+	sourceURL := u.SourceURL
+	sourceURL.User = nil
+
+	return &sourceURL, ensureGitConfigIncludes(gitconfig.Name())
```

Comment:
I'd like to see the following tests:
* No user/pw in URL, no user/pw/token in secret: no gitconfig written, no URL override returned
* User in URL, no user/pw/token in secret: no gitconfig written, no URL override returned
* User and pw in URL, no user/pw/token in secret: gitconfig written with user/pw from URL, URL override returned with user/pw stripped
* User and pw in URL, user and pw in secret: gitconfig written with user/pw from secret, URL override returned with user/pw stripped
* User in URL only, password in secret only: gitconfig written with user from URL, password from secret, URL override returned with user stripped
* No user/pw in URL, password in secret: gitconfig written with default user, password from secret